### DISCUSSION TITLE
Problem with https on Keycloak upgrade

### DISCUSSION THREAD

**Discussion by factsnobby** [keycloak/keycloak#22315] [Q&A] [answered]
Hi everyone, 

Hopefully you can give me a guiding light on what might be happening with my Keycloak installation.
I'm using the official docker image from Keycloak on version 20.0.1. We're using the Keycloak operator on a K8s setup. The operator version is matching the Keycloak version.

We have a forced redirection from _http_ to _https_. This was working fine on version 20.0.1.

Upon migrating to 22.0.1, I'm facing an issue and having trouble identifying if any of the changes included in either version 21 or 22 are having an impact. The infrastructure did not change.

After the upgrade, accessing the admin area ends in an infinitely loading spinner with the following errors visible in the browser console:

```
Refused to frame ‘http://auth.company.com/’ because it violates the following Content Security Policy directive: “frame-src ‘self’”.
Timeout when waiting for 3rd party check iframe message.
```

Also after the upgrade, also having problems with the discovery document, that is required on my Angular frontend application that uses https://www.npmjs.com/package/angular-oauth2-oidc:

```
invalid issues in discovery document expected: https://auth.company.com/realms/main current: http://auth.company.com/realms/main
init keycloak failed discovery_document_validation_error
```

In both cases it seems that Keycloak is now actively not using **https** but only **http**. 

Accessing the discovery document under https://auth.company.com/realms/main is possible with https but the _token-service_ and _account-service_ URLs contained inside the discovery document are also using _http_.

I am currently using the following environment variables to start keycloak:

```
KC_HEALTH_ENABLED=true
KC_METRICS_ENABLED=true
KC_FEATURES=account-api,update-email,impersonation,admin2,account2
KC_DB=postgres
KC_CACHE : ispn
KC_CACHE_STACK : kubernetes
KC_DB : value
KC_DB_PASSWORD : value
KC_DB_POOL_INITIAL_SIZE : value
KC_DB_POOL_MAX_SIZE : value
KC_DB_POOL_MIN_SIZE : value
KC_DB_SCHEMA : value
KC_DB_URL_DATABASE : value
KC_DB_URL_HOST : value
KC_DB_URL_PORT : value
KC_DB_USERNAME : value
KC_HEALTH_ENABLED : true
KC_HOSTNAME_STRICT : false
KC_HOSTNAME_STRICT_BACKCHANNEL : false
KC_HTTPS_PORT : 8443
KC_HTTP_ENABLED : true
KC_HTTP_PORT : 8080
KC_PROXY : passthrough
```

Does anyone have any clue of what I could try? Which change on version 21 or 22 caused issues that I've missed during my analysis?

**Comment by vmuzikar** [comment, marked as answer]
Probably related to:
https://github.com/keycloak/keycloak/blob/490016569164086a8c2040faac51ef060bf7b277/docs/documentation/upgrading/topics/keycloak/changes-22_0_0.adoc?plain=1#L175-L181

**Reply by factsnobby** [reply]
Hey @vmuzikar , we ended up in the same response in the end. Thanks for being our sanity check :) 
We are using the Keycloak Operator, so putting in the PROXY value was a bit of a challenge for us but in the end it worked with the _edge_ property.

Thank you
